errors/dynamic-server-error.mdx

```diff
@@ -0,0 +1,65 @@
+---
+title: DynamicServerError - Dynamic Server Usage
+---
+
+#### Why This Message Occurred
+
+You attempted to use a Next.js function that depends on Async Context (such as `headers` or `cookies` from `next/headers`) but it was not bound to the same call stack as the function that ran it. Next.js will throw a `DynamicServerError` to know when to opt a page into dynamic rendering. However, when it's uncaught, that means that the async context was lost.
+
+## What is Async Context?
+
+[Async Context](https://github.com/tc39/proposal-async-context) is a way to pass data within the same call stack, even through asynchronous operations. This is very useful in Next.js, where functions like cookies or headers might be called from anywhere within a React component tree or other functions during React rendering. However, this context can be lost if certain operations like setTimeout or unhandled Promise rejections are used.
+
+## Scenarios that can cause this to happen
+
+- The function was called inside of a `setTimeout` or `setInterval`, causing the value to be read outside of the call stack that the context was bound to
+- The function was called after an async operation, but the promise wasn't returned. This can cause the function to be called after the async operation has completed, resulting in a new execution context and loss of the original async context.
```

Comment:
```suggestion
- The function was called after an async operation, but the promise wasn't awaited. This can cause the function to be called after the async operation has completed, resulting in a new execution context and loss of the original async context.
```